app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/NoClaimNotificationController.scala

```diff
@@ -40,45 +44,97 @@ class NoClaimNotificationController @Inject() (
                                                 override val escService: EscService,
                                                 auditService: AuditService,
                                                 timeProvider: TimeProvider,
-                                                noClaimNotificationPage: NoClaimNotificationPage
+                                                noClaimNotificationPage: NoClaimNotificationPage,
+                                                noClaimConfirmationPage: NoClaimConfirmationPage,
 )(implicit val appConfig: AppConfig, val executionContext: ExecutionContext)
     extends BaseController(mcc)
     with LeadOnlyUndertakingSupport {
   import actionBuilders._
 
   def getNoClaimNotification: Action[AnyContent] = verifiedEmail.async { implicit request =>
     withLeadUndertaking { undertaking =>
-      val previous = routes.AccountController.getAccountPage().url
-      Ok(noClaimNotificationPage(noClaimForm, previous)).toFuture
+      retrieveSubsidies(undertaking.reference).toContext
+        .foldF(handleMissingSessionData("No claim notification - subsidies -")) { undertakingSubsidies =>
+          val previous = routes.AccountController.getAccountPage().url
+          val today = timeProvider.today
+          val startDate = today.toEarliestTaxYearStart
+
+          val lastSubmitted = undertakingSubsidies.lastSubmitted.orElse(undertaking.lastSubsidyUsageUpdt)
+
+          Ok(
+            noClaimNotificationPage(
+              noClaimForm,
+              previous,
+              undertakingSubsidies.hasNeverSubmitted, 
+              startDate.toDisplayFormat,
+              lastSubmitted.map(_.toDisplayFormat).getOrElse(""), // TODO - can we display something sensible if the value is missing?
```

Comment:
Quick question - was this meant to be fixed in this PR?

Reply:
Not yet. Something to revisit later.